fn_guardium_insights_integration/fn_guardium_insights_integration/components/funct_function_guardium_insights_classification_report.py

```diff
@@ -0,0 +1,150 @@
+            if input_field_guardium_insights_what:
+                _datasourceip = input_field_guardium_insights_what.get("server_ip")
+                _port = input_field_guardium_insights_what.get("server_port")
+
+            report_payload = gi_services.generate_report_payload(_from_date, _to_date,
+                                                                 input_field_guardium_insights_fetch_size,
+                                                                 _datasourceip,
+                                                                 _port)
+            gi_list_reports = gi_services.list_all_reports()
+            report_id = gi_services.get_report_id(gi_list_reports, "Classification")
+            if report_id:
+                # Run classification report.
+                report_data = gi_services.run_report(report_payload)
+
+                # Correcting received response, if its not correct.
+                if six.PY3:
+                    if isinstance(report_data, str):
+                        report_data = re.sub(r"{.result.*\{.total_number_of_rows.*\}", "", report_data)
+                        report_data = json.loads(report_data)
+                elif six.PY2:
+                    if isinstance(report_data, str) or isinstance(report_data, unicode):
+                        if isinstance(report_data, unicode):
+                            report_data = str(report_data)
+                        report_data = re.sub(r"{.result.*\{.total_number_of_rows.*\}", "", report_data)
+                        report_data = json.loads(report_data)
+
+                # clearing the existing table data
+                log.info("Clearing existing table data...")
+                res_dt_service.clear_existing_table_data()
+
+                # adding report data to the resilient table
+                log.info("Adding report data to the resilient table...")
+                data = report_data.get("result").get("data", {})
+                if data:
+                    cell_data_list = []
+                    for each_data in data:
+                        event = each_data.get("results")
+                        cell_data = res_dt_service.construct_table_payload(event)
+                        cell_data_list.append(cell_data)
+                    res_dt_service.add_table_data(cell_data_list)
+                    res_status = True
+                    reason = "Classification report generated successfully..."
+                else:
+                    reason = "Classification report: No results found. Try adjusting the report settings."
+                    log.info(reason)
+            else:
+                reason = "--------Classification report not found in Guardium Insight system---------"
+                log.info(reason)
+
+            yield StatusMessage("classification report completed: incident ID{}".format(incident_id))
+
+            fun_result = res_pay.done(res_status, {}, reason)
+
+            results = {
```

Comment:
In our code, we just return the results from res_pay.done() in FunctionResult. Your use of results, we make parsing the json a little more complicated.

Reply:
Done, removed additional dict.